lib/rules/require-unicode-regexp.js

```diff
@@ -44,23 +49,71 @@ module.exports = {
                 const flags = node.regex.flags || "";
 
                 if (!flags.includes("u")) {
-                    context.report({ node, messageId: "requireUFlag" });
+                    context.report({
+                        messageId: "requireUFlag",
+                        node,
+                        suggest: isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, node.regex.pattern)
+                            ? [
+                                {
+                                    fix(fixer) {
+                                        return fixer.insertTextAfter(node, "u");
+                                    },
+                                    messageId: "addUFlag"
+                                }
+                            ]
+                            : null
+                    });
                 }
             },
 
             Program() {
                 const scope = context.getScope();
+                const sourceCode = context.getSourceCode();
                 const tracker = new ReferenceTracker(scope);
                 const trackMap = {
                     RegExp: { [CALL]: true, [CONSTRUCT]: true }
                 };
 
                 for (const { node } of tracker.iterateGlobalReferences(trackMap)) {
-                    const flagsNode = node.arguments[1];
+                    const [patternNode, flagsNode] = node.arguments;
+                    const pattern = getStringIfConstant(patternNode, scope);
                     const flags = getStringIfConstant(flagsNode, scope);
 
                     if (!flagsNode || (typeof flags === "string" && !flags.includes("u"))) {
-                        context.report({ node, messageId: "requireUFlag" });
+                        context.report({
+                            messageId: "requireUFlag",
+                            node,
+                            suggest: typeof pattern === "string" && isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, pattern)
+                                ? [
+                                    {
+                                        fix(fixer) {
+                                            if (flagsNode) {
+                                                if (flagsNode.type === "Literal" && typeof flagsNode.value === "string") {
```

Comment:
```suggestion
                                                if ((flagsNode.type === "Literal" && typeof flagsNode.value === "string") || flagsNode.type === "TemplateLiteral") {
```

We can include template literals here.